src/libmongoc/src/mongoc/mongoc-cluster.c

```diff
@@ -3512,9 +3513,10 @@ _mongoc_cluster_run_opmsg_send (mongoc_cluster_t *cluster,
 
    mongoc_server_stream_t *const server_stream = cmd->server_stream;
 
-   const uint32_t flags = cmd->is_acknowledged
-                             ? MONGOC_OP_MSG_FLAG_NONE
-                             : MONGOC_OP_MSG_FLAG_MORE_TO_COME;
+   const uint32_t flags =
+      (cmd->is_acknowledged ? MONGOC_OP_MSG_FLAG_NONE
+                            : MONGOC_OP_MSG_FLAG_MORE_TO_COME) |
+      (cmd->op_msg_is_exhaust ? MONGOC_OP_MSG_FLAG_EXHAUST_ALLOWED : 0);
```

Comment:
```suggestion
      (cmd->op_msg_is_exhaust ? MONGOC_OP_MSG_FLAG_EXHAUST_ALLOWED
                              : MONGOC_OP_MSG_FLAG_NONE);
```
Consistency + avoid implicit integral conversion/promotion.